##Somatic

**Varscan:**
The algorithm paper is available in [VarScan 2: Somatic mutation and copy number alteration discovery in cancer by exome sequencing](http://genome.cshlp.org/content/22/3/568.short)	

	Tumor-normal pair samples, but mainly for somatic mutation detection
	dataset used: ovarian tumor
Dataset link: 

[ovarian tumor](http://www.ncbi.nlm.nih.gov/gap)

[googled ovarian tumor data TCGA](http://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/study.cgi?study_id=phs000178.v3.p3)

**Strelka:**
[Strelka somatic variant caller - Google Sites](https://sites.google.com/site/strelkasomaticvariantcaller/) Strelka is an analysis package designed to detect somatic SNVs and small indels from the aligned sequencing reads of matched tumor-normal samples. 

	Tumor-normal pair, but mainly for somatic SNV and small indel detection
	dataset used: ovarian tumor

Website:[DATASET: A Comprehensive Catalogue of Somatic Mutations from a Human Cancer Genome](https://www.ebi.ac.uk/ega/datasets/EGAD00001000149)
**No acess download**


**Mutect:**
[Sensitive detection of somatic point mutations in impure and heterogeneous cancer samples](http://www.nature.com/nbt/journal/v31/n3/abs/nbt.2514.html)
MuTect is a method that applies a Bayesian classifier to detect somatic mutations with very low allele fractions, requiring only a few supporting reads, followed by carefully tuned filters that ensure high specificity.
[**Virtual** dataset](ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/technical/working/20120117_ceu_trio_b37_decoy/)


**Cake:**
[Cake: a bioinformatics pipeline for the integrated analysis of somatic variants in cancer genomes](http://bioinformatics.oxfordjournals.org/content/29/17/2208.full.pdf+html)
Bioinformatics (2013) 29 (17): 2208-2210.
doi: 10.1093/bioinformatics/btt371
First published online: June 25, 2013

matched tumor-normal samples
data information in [supple info](http://bioinformatics.oxfordjournals.org/content/suppl/2013/06/23/btt371.DC1/20130606_CAKE_Supp_Information.pdf)

Human hepatocellular carcinoma

breast cancer

**Mutascope:**

matched tumor-normal specimen
**virtual** dataset
The data are available through the Short Reads Archive
(SRA) at the NCBI (SRA067609 and SRA067610).
or breast cancer?

[Mutascope: sensitive detection of somatic mutations from deep amplicon sequencing](http://bioinformatics.oxfordjournals.org/content/29/15/1908.full.pdf+html)
Bioinformatics (2013) 29 (15): 1908-1909.
doi: 10.1093/bioinformatics/btt305
First published online: May 27, 2013

**Shimmer:**
[Shimmer: detection of genetic alterations in tumors using next-generation sequence data](http://bioinformatics.oxfordjournals.org/content/29/12/1498.full.pdf+html)
Bioinformatics (2013) 29 (12): 1498-1503.
doi: 10.1093/bioinformatics/btt183
First published online: April 24, 2013

You may access this article for 1 day for **US$35.00.**


[An empirical Bayesian framework for somatic mutation detection from cancer genome sequencing data](http://nar.oxfordjournals.org/content/41/7/e89.full.pdf+html)
Nucl. Acids Res. (2013) 41 (7): e89.
doi: 10.1093/nar/gkt126
First published online: March 6, 2013

After receiving informed consent, paired tumour-normal samples were obtained from 20 patients with clear cell renal cell carcinoma (**ccRCC**) by sampling their specimens
during surgical operations

[A survey of tools for variant analysis of next-generation genome sequencing data](http://bib.oxfordjournals.org/content/early/2013/01/21/bib.bbs086.full.pdf+html)
Brief Bioinform (2013)
doi: 10.1093/bib/bbs086
First published online: January 21, 2013
For testing SVs, a whole-genome sequencing data from a **liver metastatic lung cancer samples** obtained from lung adenocarcinoma patients (ERP001071,[95]) was selected (WGS). We randomly chose one tumor/normal pair (subject AK55) sequenced on an Illumina HiSeq 2000 of the available data sets.

[Lung adenocarcinoma patients AK55 ERP001071](http://sra.dnanexus.com/studies/ERP001071)

[Download](http://sra.dnanexus.com/dispatch_many?accession_type=Study&accessions%5B%5D=ERP001071&action_name=download "Download")

**Virmid**: 
[Virmid: accurate detection of somatic mutations with sample impurity inference.](http://genomebiology.com/content/14/8/R90) **$32**

hemimegalencephaly exome sequencing data
Data reference [De novo somatic mutations in components of the PI3K-AKT3-mTOR pathway cause hemimegalencephaly](http://www.nature.com/ng/journal/v44/n8/full/ng.2329.html) 

##Virus
[Complete viral RNA genome sequencing of ultra-low copy samples by sequence-independent amplification.](http://www.ncbi.nlm.nih.gov/pubmed/22962364)
Dataset information:

	HIV dataset
	HIV clone and HIV clinical sample
	in HIV folder

[LoFreq: a sequence-quality aware, ultra-sensitive variant caller for uncovering cell-population heterogeneity from high-throughput sequencing datasets](http://nar.oxfordjournals.org/content/40/22/11189.full.pdf+html?sid=0d1dbba7-bea1-414a-92a9-823e532748c1)
Nucl. Acids Res. (2012) 40 (22): 11189-11201.
doi: 10.1093/nar/gks918
First published online: October 12, 2012
	
	not normal-tumor pair
	DENV2 six library replicates
data link not working [Lofreq dataset](http://collaborations.gis.a-star.edu.sg/~wilma/lofreq_paper_data/)
[http://trace.ddbj.nig.ac.jp/DRASearch/submission?acc=ERA000206](http://trace.ddbj.nig.ac.jp/DRASearch/submission?acc=ERA000206)



[Use of Four Next-Generation Sequencing Platforms to Determine HIV-1 Coreceptor Tropism](http://www.hiv.lanl.gov/content/sequence/HIV/NextGenArchive/Archer2012.html)

	12 HIV clinical sample without control
	reference: Sequences corresponding to the V3 region of the HIV surface glycoprotein (gp120) spanning positions 7,110 to 7,217 on the HXB2 reference strain (accession No. K03455) were extracted and aligned generating 48 nucleotide alignments, i.e., 12 samples x 4 NGS platforms.



Tumor-Normal PCR-Free WGS Data (HCC1187)
[Tumor-Normal PCR-Free WGS Data (HCC1187)](http://blog.basespace.illumina.com/2013/04/11/tumor-normal-pcr-free-wgs-data-hcc1187/)

[Molecular Characterization of Tumors Using 
Next-Generation Sequencing 
](http://res.illumina.com/documents/products/technotes/technote_molecular_characterizations_of_tumors.pdf)

[TumorNormal_WGS_HiSeq2000_CSW_0.23](https://basespace.illumina.com/project/744744/TumorNormal_WGS_HiSeq2000_CSW_0-23#/samples)

Materials and Methods: The DNA was extracted from the HCC1187 breast ductal carcinoma cell line and a matching lymphoblastoid cell line.

H1N1 (A/H1N1/09) influenza
[Viral population analysis and minority-variant detection using short read next-generation sequencing.](http://www.ncbi.nlm.nih.gov/pubmed/23382427)