##Somatic

**Varscan:**
The algorithm paper is available in [VarScan 2: Somatic mutation and copy number alteration discovery in cancer by exome sequencing](http://genome.cshlp.org/content/22/3/568.short)	

	Tumor-normal pair samples, but mainly for somatic mutation detection
	dataset used: ovarian tumor

**Strelka:**
[Strelka somatic variant caller - Google Sites](https://sites.google.com/site/strelkasomaticvariantcaller/) Strelka is an analysis package designed to detect somatic SNVs and small indels from the aligned sequencing reads of matched tumor-normal samples. 
	Tumor-normal pair, but mainly for somatic SNV and small indel detection
	dataset used: ovarian tumor

**Mutect:**
[Sensitive detection of somatic point mutations in impure and heterogeneous cancer samples](http://www.nature.com/nbt/journal/v31/n3/abs/nbt.2514.html)
MuTect is a method that applies a Bayesian classifier to detect somatic mutations with very low allele fractions, requiring only a few supporting reads, followed by carefully tuned filters that ensure high specificity.

**Cake:**
[Cake: a bioinformatics pipeline for the integrated analysis of somatic variants in cancer genomes](http://bioinformatics.oxfordjournals.org/content/29/17/2208.full.pdf+html)
Bioinformatics (2013) 29 (17): 2208-2210.
doi: 10.1093/bioinformatics/btt371
First published online: June 25, 2013

**Mutascope:**
[Mutascope: sensitive detection of somatic mutations from deep amplicon sequencing](http://bioinformatics.oxfordjournals.org/content/29/15/1908.full.pdf+html)
Bioinformatics (2013) 29 (15): 1908-1909.
doi: 10.1093/bioinformatics/btt305
First published online: May 27, 2013

**Shimmer:**
[Shimmer: detection of genetic alterations in tumors using next-generation sequence data](http://bioinformatics.oxfordjournals.org/content/29/12/1498.full.pdf+html)
Bioinformatics (2013) 29 (12): 1498-1503.
doi: 10.1093/bioinformatics/btt183
First published online: April 24, 2013

[An empirical Bayesian framework for somatic mutation detection from cancer genome sequencing data](http://nar.oxfordjournals.org/content/41/7/e89.full.pdf+html)
Nucl. Acids Res. (2013) 41 (7): e89.
doi: 10.1093/nar/gkt126
First published online: March 6, 2013

[A survey of tools for variant analysis of next-generation genome sequencing data](http://bib.oxfordjournals.org/content/early/2013/01/21/bib.bbs086.full.pdf+html)
Brief Bioinform (2013)
doi: 10.1093/bib/bbs086
First published online: January 21, 2013
For testing SVs, a whole-genome sequencing data from a **liver metastatic lung cancer samples** obtained from lung adenocarcinoma patients (ERP001071,[95]) was selected (WGS). We randomly chose one tumor/normal pair (subject AK55) sequenced on an
Illumina HiSeq 2000 of the available data sets.



##Virus
[Complete viral RNA genome sequencing of ultra-low copy samples by sequence-independent amplification.](http://www.ncbi.nlm.nih.gov/pubmed/22962364)
Dataset information:

	HIV dataset
	HIV clone and HIV clinical sample
	in HIV folder

[LoFreq: a sequence-quality aware, ultra-sensitive variant caller for uncovering cell-population heterogeneity from high-throughput sequencing datasets](http://nar.oxfordjournals.org/content/40/22/11189.full.pdf+html?sid=0d1dbba7-bea1-414a-92a9-823e532748c1)
Nucl. Acids Res. (2012) 40 (22): 11189-11201.
doi: 10.1093/nar/gks918
First published online: October 12, 2012

	DENV2 six library replicates

[Use of Four Next-Generation Sequencing Platforms to Determine HIV-1 Coreceptor Tropism](http://www.hiv.lanl.gov/content/sequence/HIV/NextGenArchive/Archer2012.html)
	12 HIV clinical sample without control
