Massively parallel sequencing data generated by next-generation sequencing (NGS) technology is routinely used to detect single nucleotide variants (SNVs) in research samples. 

An emerging challenge for this technology is the identification of SNVs in heterogeneous cell populations with low read-depth data.

We have developed a Bayesian statistical model is able to share information between correlated positions and call low-frequency variants in heterogeneous samples. 

We present a Bayesian sensitivity analysis of the model to variations in the prior function. 

A Jeffreys prior gives a lower false discovery rate (FDR) to detect a 0.1% minor allele frequency event compared with an improper prior. 

In an analysis of a directed evolution experiment, we are able to call mutations with high sensitivity and specificity, and detect the emergence of a beneficial SNV earlier than was previously shown.

