2013-09-21 SNP calling using virmid
==============================

Purpose
------------
To compare virmid with rvd2 on variant calling

Conclusions
-----------------
This approach is not able to run on our dataset for unknow reason.

Background
----------------
Virmid paper is available in Genome Biology [Virmid: accurate detection of somatic mutations with sample impurity inference](http://genomebiology.com/2013/14/8/R90/abstract)

Software implementation webpage[varmid: Virtual Microdissection for SNP calling](http://sourceforge.net/projects/virmid/)

Virmid description from its manual page:

Virmid (Virtual Microdissection for SNP calling) is a Java based variant caller designed for disease-control matched samples. Virmid is also specialized for identifying potential within individual contamination where the disease sample cannot be purified enough. While the SNP calling rate is severely compromised with this heterogeneity, Virmid can uncover SNPs with low allele frequency by considering the level of contamination (alpha). The important features of Virmid are:

•Estimation of accurate proporation of control sample in a (mixed) disease sample

•Improved SNP and somatic mutation calling with regard to the estimated proportion


Materials and Equipment
------------------------------



Experimental Protocol
---------------------------
A sample experiment is implemented in `SNP_virmid.sh`

Results
-----------


Archived Samples
-------------------------

Archived Computer Data
------------------------------


Prepared by: _____Yuting He____________     Date: _______09/30/13______________


Witnessed by: ________________________
